presto-main/src/test/java/com/facebook/presto/type/AbstractTestType.java

```diff
@@ -257,6 +286,43 @@ else if (type.getJavaType() == Slice.class) {
             }
             catch (IllegalStateException | UnsupportedOperationException expected) {
             }
+            try {
+                type.getObject(block, position);
+                fail("Expected IllegalStateException or UnsupportedOperationException");
+            }
+            catch (IllegalStateException | UnsupportedOperationException expected) {
+            }
+        }
+        else {
+            SliceOutput actualSliceOutput = new DynamicSliceOutput(100);
+            BlockSerdeUtil.writeBlock(actualSliceOutput, (Block) type.getObject(block, position));
+            SliceOutput expectedSliceOutput = new DynamicSliceOutput(actualSliceOutput.size());
+            BlockSerdeUtil.writeBlock(expectedSliceOutput, (Block) expectedStackValue);
+            assertEquals(actualSliceOutput.slice(), expectedSliceOutput.slice());
```

Comment:
wouldn't it be more safe to put these into `Block`s and then assert `Type.equalTo`?